src/main/java/com/microsoft/jenkins/appservice/WebAppSwapSlotsRecorder.java

```diff
@@ -0,0 +1,190 @@
+package com.microsoft.jenkins.appservice;
+
+import com.microsoft.azure.PagedList;
+import com.microsoft.azure.management.Azure;
+import com.microsoft.azure.management.appservice.DeploymentSlot;
+import com.microsoft.azure.management.appservice.WebApp;
+import com.microsoft.jenkins.appservice.util.AzureUtils;
+import com.microsoft.jenkins.appservice.util.Constants;
+import com.microsoft.jenkins.exceptions.AzureCloudException;
+import hudson.AbortException;
+import hudson.Extension;
+import hudson.FilePath;
+import hudson.Launcher;
+import hudson.model.Item;
+import hudson.model.Run;
+import hudson.model.TaskListener;
+import hudson.tasks.BuildStepMonitor;
+import hudson.tasks.Recorder;
+import hudson.util.FormValidation;
+import hudson.util.ListBoxModel;
+import jenkins.tasks.SimpleBuildStep;
+import org.apache.commons.lang3.StringUtils;
+import org.jenkinsci.Symbol;
+import org.kohsuke.stapler.AncestorInPath;
+import org.kohsuke.stapler.DataBoundConstructor;
+import org.kohsuke.stapler.QueryParameter;
+
+import javax.annotation.Nonnull;
+import java.io.IOException;
+import java.util.NoSuchElementException;
+
+public class WebAppSwapSlotsRecorder extends Recorder implements SimpleBuildStep {
+    private final String azureCredentialsId;
+    private final String resourceGroup;
+    private final String appName;
+    private final String sourceSlotName;
+    private final String targetSlotName;
+
+    @DataBoundConstructor
+    public WebAppSwapSlotsRecorder(String azureCredentialsId,
+                                   String resourceGroup,
+                                   String appName,
+                                   String sourceSlotName,
+                                   String targetSlotName) {
+        this.azureCredentialsId = azureCredentialsId;
+        this.resourceGroup = resourceGroup;
+        this.appName = appName;
+        this.sourceSlotName = sourceSlotName;
+        this.targetSlotName = targetSlotName;
+    }
+
+    public String getAzureCredentialsId() {
+        return azureCredentialsId;
+    }
+
+    public String getResourceGroup() {
+        return resourceGroup;
+    }
+
+    public String getAppName() {
+        return appName;
+    }
+
+    public String getSourceSlotName() {
+        return sourceSlotName;
+    }
+
+    public String getTargetSlotName() {
```

Comment:
preserve virtual network?

Reply:
There is no direct method or parameters can controller this as the REST api does. Have create an issue for the sdk team at Azure/azure-libraries-for-java#742 .